cirq-core/cirq/transformers/dynamical_decoupling.py

```diff
@@ -276,14 +276,24 @@ def _update_pulled_through(q: ops.Qid, insert_gate: ops.Gate) -> ops.Operation:
             dd_iter_by_qubits[q] = cycle(base_dd_sequence)
         # Need to insert a new moment before current moment
         if new_moment_ops:
-            # Fill insertable idle moments in the new moment using dd sequence
-            for q in orig_circuit.all_qubits() - stop_pulling_through_qubits:
-                if busy_moment_range_by_qubit[q][0] < moment_id <= busy_moment_range_by_qubit[q][1]:
-                    new_moment_ops.append(_update_pulled_through(q, next(dd_iter_by_qubits[q])))
             moments_to_be_appended = _try_merge_single_qubit_ops_of_two_moments(
-                transformed_moments.pop(), circuits.Moment(new_moment_ops)
+                transformed_moments[-1], circuits.Moment(new_moment_ops)
             )
-            transformed_moments.extend(moments_to_be_appended)
+            if len(moments_to_be_appended) == 1:
+                transformed_moments.pop()
```

Comment:
These 4 lines are added to merge new_moment with prev moment first if possible, other wise.

The remaining code are unchanged except some format changes.